scripts/cli/reportI18nStats.mjs

```diff
@@ -0,0 +1,87 @@
+/// @ts-check
+
+import { readdir, stat, readFile } from 'fs/promises';
+import path from 'path';
+
+const LOCALES_DIR = path.join('.', 'public', 'locales');
+
+const locales = await readdir(LOCALES_DIR);
+
+/**
+ * @type {Array<{ language: string, untranslatedCount: number, translatedCount: number }>}
+ */
+const stats = [];
+
+for (const fileName of locales) {
+  const filePath = path.join(LOCALES_DIR, fileName, 'grafana.json');
+  if (!(await exists(filePath))) {
+    continue;
+  }
+
+  const messages = await readFile(filePath);
+  const parsedMessages = JSON.parse(messages.toString());
+
+  let translatedCount = 0;
+  let untranslatedCount = 0;
+
+  eachMessage(parsedMessages, (value) => {
+    if (value === '') {
+      untranslatedCount += 1;
+    } else {
+      translatedCount += 1;
+    }
+  });
+
+  stats.push({
+    language: fileName,
+    translatedCount,
+    untranslatedCount,
+  });
+}
+
+for (const stat of stats) {
+  logStat(`untranslated.${stat.language}`, stat.untranslatedCount);
+  logStat(`translated.${stat.language}`, stat.translatedCount);
+}
+
+/**
+ * @param {string} filePath
+ */
+async function exists(filePath) {
+  try {
+    await stat(filePath);
+    return true;
+  } catch (err) {
+    if (err.code === 'ENOENT' || err.code === 'ENOTDIR') {
+      return false;
+    }
+    throw err;
+  }
+}
+
+/**
+ * @param {unknown} value
+ * @param {(v: string) => void} callback
+ */
+function eachMessage(value, callback) {
+  if (typeof value === 'object') {
+    for (const key in value) {
+      const element = value[key];
+      eachMessage(element, callback);
+    }
+  } else if (typeof value === 'string') {
+    callback(value);
+  } else {
+    throw new Error(`Unknown value ${value} in eachMessage`);
```

Comment:
Is it intended to fail the script here? do you want to put some kind of try/catch where you call this function?